### PR TITLE
Plumb the use of event number for MTRBaseDevice read/subscribe

### LINKED ISSUE #26184: [Darwin] MTRBaseDevice read/subscribe should make use of event number in params
The `minEventNumber` value in `MTRReadParams` and `MTRSubscribeParams` is not being used to populate `ReadPrepareParams` for ReadClient.

### PR DESCRIPTION
**Pull request by jtung-apple**
Fixes #26184

### CODE REVIEW

File: src/darwin/Framework/CHIP/MTRBaseDevice.mm [hunk 1]
```diff
@@ -364,6 +364,10 @@ - (void)subscribeWithQueue:(dispatch_queue_t)queue
                                       readParams.mAttributePathParamsListSize = 1;
                                       readParams.mpEventPathParamsList = eventPath.get();
                                       readParams.mEventPathParamsListSize = 1;
+                                      if (params.minEventNumber) {
+                                          readParams.mEventNumber.SetValue(
+                                              static_cast<chip::EventNumber>(params.minEventNumber.unsignedLongLongValue));
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRBaseDevice.mm:369]: Isn't this handled by the `[params toReadPrepareParams:readParams];` bit above?

File: src/darwin/Framework/CHIP/MTRBaseDevice.mm [hunk 2]
```diff
@@ -1007,6 +1011,9 @@ - (void)readAttributePaths:(NSArray<MTRAttributeRequestPath *> * _Nullable)attri
             readParams.mAttributePathParamsListSize = [attributePaths count];
             readParams.mpEventPathParamsList = eventPathParamsList.Get();
             readParams.mEventPathParamsListSize = [eventPaths count];
+            if ([eventPaths count] && params.minEventNumber) {
+                readParams.mEventNumber.SetValue(static_cast<chip::EventNumber>(params.minEventNumber.unsignedLongLongValue));
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRBaseDevice.mm:1015]: Again `toReadPrepareParams` handles this, I thought.

File: src/darwin/Framework/CHIP/MTRBaseDevice.mm [hunk 3]
```diff
@@ -1437,6 +1444,10 @@ - (void)subscribeToAttributePaths:(NSArray<MTRAttributeRequestPath *> * _Nullabl
                    readParams.mAttributePathParamsListSize = attributePathSize;
                    readParams.mpEventPathParamsList = container.eventPathParams;
                    readParams.mEventPathParamsListSize = eventPathSize;
+                   if (eventPathSize && params.minEventNumber) {
+                       readParams.mEventNumber.SetValue(
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRBaseDevice.mm:1448]: And here.

**Comment by jtung-apple** [comment]
Okay Boris you're right about this already being there and I just missed it. Closing both PR and issue.

**Pull request closed without merging**